### ISSUE TITLE
define() ignores namespaces

### ISSUE THREAD

**Issue by bugreportuser** [vimeo/psalm#1375]
Issue for https://github.com/vimeo/psalm/pull/1358.

Psalm ignores namespaces for `define()`. Removing the namespace was done in https://github.com/vimeo/psalm/pull/1358/files#diff-8d161e8eca0edad572c2b6d0818a4bc2R1486, but lookup needs to be changed before that code can be removed.

All these `echo` lines should cause errors (https://getpsalm.org/r/2f026abf41):
```php
<?php
namespace {
define('ns1\\cons1', 0);
}
namespace ns1 {
define(__NAMESPACE__.'\\cons2', 0);

echo \ns2\cons1;
echo ns2\cons1;
echo ns1\cons1;

echo \ns2\cons2;
echo ns2\cons2;
echo ns1\cons2;
}
namespace ns2 {
echo \ns2\cons1;
echo ns1\cons1;
}
namespace {
echo \ns2\cons1;
echo ns2\cons1;
}
```

Output:
```
Psalm output (using commit 1e2c290):

No issues!
```
